### DISCUSSION TITLE
Evolution of @prepare

### DISCUSSION THREAD

**Discussion by kardagan** [Codeception/Codeception#6198] [Ideas]
Hello,
I've got an idea to make an evolution of the @prepare annotation
I have a use case where I would like to pass parameters to the prepare that i call and have the result

so I've thinks to this

```
/**
 * @prepare useBrowser ($name = 'chrome') as webdriver
 */
public function chromeSpecificTest( $webdriver )
{
    // ...
}

protected function useBrowser(\Codeception\Module\WebDriver $webdriver, $name)
{
    // WebDriver was injected by the class name
    $webdriver->_reconfigure(['browser' => $name]);
    return $webdriver
}
```

or other use case

```
/**
 * @prepare buildData (version = 1) as data
 */
public function myTest( \FunctionalTester $I, $data )
{
    // ...
}

protected function buildData($version)
{
    return [
    	'foo' => 'bar',
    	'version' => $version
    ];
}
```

what do you think of that ? 
I'll do the PR if you're ok.

**Comment by DavertMik** [comment]
Well, that's quite a complex feature with programming in annotations. I don't think annotation magic should go this far. Any alternative ideas to solve this?

**Comment by kardagan** [comment]
I've think of it and don't find.

but for me It's not more complexe than doctrine anonnation :
https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/annotations-reference.html#annref_joincolumns

I did think about create new annotation but with `@prapare` and `@dataProvider` existing a third one may be too much...

and if I do it the basic uses will still work, it will only optional addone
